### PR TITLE
[feature] replace native color input by paper-swatch-picker

### PR DESCRIPTION
**Pull request by syu93**
Replacing native color input by paper-swatch-picker

### CODE REVIEW

**Comment by syu93** [comment]
I signed it!

**Comment by notwaldorf** [comment]
Let me first fix up the bit where I shouldn't check in the `bower_components` into this repo, then I'll ask you for a rebase (which will also make the diff more manageable 😅)

**Comment by notwaldorf** [comment]
@syu93 it's ready for a rebase!

**Comment by syu93** [comment]
Hope that will do the trick, first time contribution here :sweat_smile:,
If it's not good, I'll do my best to fix my mistakes.

Should I add the web-animation polyfill in another PR ?

**Comment by notwaldorf** [comment]
This looks great! I think you can add the polyfill in this PR, since it's needed by the `paper-swatch-picker` :)

**Comment by notwaldorf** [comment]
Oh hey, and congrats on your first PR!!! 🎉🎉🎉

**Comment by syu93** [comment]
Yeah ! Feel pretty awesome  :tada:

**Comment by syu93** [comment]
I've added the lite version of the polyfill, tested on Chrome(deb) 60 and Firefox(deb) 52

**Comment by syu93** [comment]
Is there a reason this PR has not been merged yet ?

**Comment by notwaldorf** [comment]
@syu93 I was travelling, then on vacation, and haven't had a chance to look at it. :(

File: bower.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "iron-iconset-svg": "PolymerElements/iron-iconset-svg#^2.0.0",
     "iron-selector": "PolymerElements/iron-selector#^2.0.0",
     "iron-pages": "PolymerElements/iron-pages#^2.0.0",
-    "iron-ajax": "PolymerElements/iron-ajax#^2.0.0"
+    "iron-ajax": "PolymerElements/iron-ajax#^2.0.0",
+    "web-animations-js": "^2.3.1"
```
Comment (notwaldorf) [inline bower.json]: `paper-swatch-picker` needs to be here

File: elements/element-stuff-base.html [hunk 2]
```diff
@@ -16,13 +16,14 @@
...
           this.root.addEventListener('change', this._do.bind(this));
+          this.root.addEventListener('color-picker-selected', this._do.bind(this));
         }
 
         _recomputeStuff() {
           this._stuff = [];
           let els = this.root.querySelectorAll('[name]');
           for (let i = 0 ; i < els.length; i++) {
-            this._stuff.push(els[i].name);
+            this._stuff.push(els[i].getAttribute('name'));
```
Comment (notwaldorf) [inline elements/element-stuff-base.html]: Why is this change needed?

File: elements/element-stuff-base.html [hunk 3]
```diff
@@ -37,11 +38,17 @@
             return;
           }
 
+          // FIXME : A bit silly but useful for custom-element cases
+          let _value = target.value;
```
Comment (notwaldorf) [inline elements/element-stuff-base.html]: Instead of this, I think there should be a special class name on the colour/background colour inputs, so that we can write an `if` statement similar to the one above.

Comment (notwaldorf) [reply elements/element-stuff-base.html]: or, `let value = target.value || target.color`, but i'm worried that if the `value` is false or undefined it would fallback to the `color` incorrectly

File: elements/element-stuff-base.html [hunk 4]
```diff
@@ -37,11 +38,17 @@
...
+          if (!_value) {
+            _value = target.color;
+          }
+
           Polymer.Base.fire('element-updated',
               {
                 type: this.stuffType,
-                name: target.name,
-                value: target.value,
+                name: target.getAttribute('name'),
```
Comment (notwaldorf) [inline elements/element-stuff-base.html]: why is this change needed?

File: elements/element-stuff-base.html (hunk 2 above)
Comment (syu93) [reply elements/element-stuff-base.html]: Hi, I'm sorry, I've been really busy lately with my last project :confused:
I was not sure about this change, but it seems that, as the  `name` attribute is not a property of `paper-swatch-picker`, we can't access it directly, that's why I needed to use it as attribute

File: elements/element-stuff-base.html (hunk 3 above)
Comment (syu93) [reply elements/element-stuff-base.html]: The second is a more elegant solution, but I'll go with the if statement to avoid possible issue here

File: elements/element-stuff-base.html (hunk 4 above)
Comment (syu93) [reply elements/element-stuff-base.html]: this is linked to the previous issue with the name as an attribute

**Comment by syu93** [comment]
Hi, sorry to bother you, I know you're pretty busy, Am I all wrong on this PR ? Or is it not relevant anymore ? In that case maybe we should close it

**Comment by notwaldorf** [comment]
i just looked at it for a review, but it needs a rebase (i can't run it locally)

**Comment by notwaldorf** [comment]
(the PR looks otherwise fine to merge, I think!)

**Comment by syu93** [comment]
This should be good right now,
Pls tell if anything goes wrong, hope I finally gonna have my first PR merged ^^"

**Comment by notwaldorf** [comment]
Looks good! Thanks so much for sticking through with this! 🎉🎉🎉
